matter_server/common/helpers/util.py

```diff
@@ -0,0 +1,228 @@
+"""Utils for Matter server (and client)."""
+from base64 import b64encode
+from dataclasses import MISSING, asdict, dataclass, fields, is_dataclass
+from datetime import date, datetime
+from enum import Enum
+from functools import cache
+from importlib.metadata import PackageNotFoundError, version as pkg_version
+import logging
+import platform
+from typing import Any, Dict, Optional, Set, Type, Union, get_args, get_origin
+
+from chip.clusters import Cluster, ClusterObject
+from chip.clusters.Types import Nullable, NullValue
+from chip.tlv import float32, uint
+import pkg_resources
+
+try:
+    # python 3.10
+    from types import NoneType
+except:  # noqa
+    # older python version
+    NoneType = type(None)
+
+CHIP_CLUSTERS_PKG_NAME = "home-assistant-chip-clusters"
+CHIP_CORE_PKG_NAME = "home-assistant-chip-core"
+
+# TODO: The below dataclass utils are shamelessly copied from aiohue
```

Comment:
yeah, I'm looking at that now because this still does not feel right. I have another serialization issue so I might end up just using dacite because all models are dataclasses.

Reply:
too many weird value types so using dacite turned out to be a nightmare. The custom code is now (finally) working well for everything I have tested so far and its now able to dynamically serialize to/from the data class models of the clusters package, including the weird stuff with like actually class types as dict keys.

We can improve and cleanup later but for now it works. Spent too many day trying to get it to work with existing helper packages.